axelor-sale/src/main/java/com/axelor/apps/sale/rest/SaleOrderRestController.java

```diff
@@ -63,4 +71,25 @@ public Response createSaleOrder(SaleOrderPostRequest requestBody)
 
     return ResponseConstructor.buildCreateResponse(saleOrder, new SaleOrderResponse(saleOrder));
   }
+
+  @Operation(
+      summary = "Update sale order status",
+      tags = {"Sale order"})
+  @Path("/status")
+  @PUT
+  @HttpExceptionHandler
+  public Response changeSaleOrderStatus(SaleOrderPutRequest requestBody) throws AxelorException {
+    RequestValidator.validateBody(requestBody);
+    new SecurityCheck().writeAccess(SaleOrder.class, requestBody.getSaleOrderId()).check();
+    String toStatus = requestBody.getToStatus();
+    SaleOrder saleOrder = requestBody.fetchSaleOrder();
+    if (Objects.equals(toStatus, SaleOrderPutRequest.SALE_ORDER_UPDATE_FINALIZE)) {
```

Comment:
Please use SaleOrderPutRequest.SALE_ORDER_UPDATE_FINALIZE.equals(toStatus) instead of Objects.equals